src/include/sof/audio/component.h

```diff
@@ -424,6 +429,20 @@ static inline int comp_params(struct comp_dev *dev,
 	return 0;
 }
 
+/**
+ * Fetch hardware stream parameters - only mandatory for DAI components.
+ * @param dev Component device.
+ * @param params hw parameters
+ * @return 0 if succeeded, error code otherwise.
+ */
+static inline int comp_dai_get_hw_params(struct comp_dev *dev,
+					 struct sof_ipc_stream_params *params)
+{
+	if (dev->drv->ops.dai_get_hw_params)
+		return dev->drv->ops.dai_get_hw_params(dev, params);
+	return 0;
```

Comment:
I wonder whether it's better to return success or error when this method is called on non-DAI components.

Reply:
Lets return an error here. This will at least tell callers there is no params data too.